docs/upgrading.asciidoc

```diff
@@ -7,12 +7,33 @@ endif::[]
 
 == Upgrading
 
+Upgrades between minor versions of the agent, like from 1.1 to 1.2 are always backwards compatible.
+Upgrades that involve a major version bump often come with some backwards incompatible changes.
```

Comment:
I would probably add the following sentence before the two you just added:

> The Elastic APM Node.js Agent uses [semantic versioning](https://semver.org/), and therefore:

Is there a reason you say "often" and not "always" in the last sentence?

Reply:
Will do!

> Is there a reason you say "often" and not "always" in the last sentence?

Just trying to be safe. I can update to _always_ if you'd like.

Reply:
Nah, it's a more soft wording to use "often" (though technically there's no reason to make a new major unless there's a breaking change, so technically there will always be one in that case - but I'm fine with "often")